src/lib/data-cache.ts

```diff
@@ -294,7 +295,7 @@ export async function cacheManyFromCdnWithProgress(
         });
 
         if (batchIds.length > 0) {
-            const isMetadata = idsToUrls[0]?.url.includes('metadata');
+            const isMetadata = objectValues(idsToUrls)[0]?.url.includes('metadata') ?? false;
```

Comment:
Threw in this fix as well, it fixes a bug from the batching refactor that caused the download manager to not cache metadata.